### PR TITLE
feat: 13412 - improve import app with progress bar

### LINKED ISSUE #13412: [Feature]-[300]:Improve import application UX flow
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

As per the current implementation, there is no loading state-provided when a user tries to import the application. This results in a bad UX when we have a large application that takes more than 10sec to import the app successfully.   

Checkout the current experience:
[LOOM DEMO](https://www.loom.com/share/f317f4bc453f4963859fea41ad96325a)

Design handoff: https://www.figma.com/file/IOx0WPeZsChCaXwgVxU3LX/Import%2FExport-Design?node-id=1594%3A8197
Screenshot 2022-04-28 at 3 48 13 PM

### Why should this be worked on?

This will improve the import application UX especially for larger apps.

### PR DESCRIPTION
**Pull request by haojin111**
## Description

> Added status bar while importing application with json

Fixes #13412 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Jest test
> cypress test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: feat/13412-improve-import-app-flow-ui 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.63 **(0.01)** | 38.63 **(0)** | 35.97 **(0.03)** | 56.88 **(0.01)**
 :green_circle: | app/client/src/ce/constants/messages.ts | 77.99 **(0.21)** | 100 **(0)** | 34.88 **(0.62)** | 82.03 **(0.29)**</details>

### CODE REVIEW

**Review by IAmAnubhavSaini — CHANGES_REQUESTED** [review]
Remove coverage-summary.json from changeset.

**Comment by IAmAnubhavSaini** [comment]
There are 0 files to review.

File: app/client/src/ce/constants/messages.test.ts [hunk 1]
```diff
@@ -386,3 +392,46 @@ describe("git-sync messages", () => {
     });
   });
 });
+
+describe("import application messages", () => {
+  const expectedMessages = [
+    { key: "UPLOADING_JSON", value: "Uploading JSON" },
```
Comment (IAmAnubhavSaini) [inline app/client/src/ce/constants/messages.test.ts:398]: Should be `Uploading JSON file`.

File: app/client/src/ce/constants/messages.test.ts [hunk 2]
```diff
@@ -386,3 +392,46 @@ describe("git-sync messages", () => {
...
+    {
+      key: "UPLOADING_APPLICATION",
+      value: "Uploading Application",
```
Comment (IAmAnubhavSaini) [inline app/client/src/ce/constants/messages.test.ts:401]: Should be `Uploading application`.

File: app/client/src/ce/constants/messages.test.ts [hunk 3]
```diff
@@ -386,3 +392,46 @@ describe("git-sync messages", () => {
...
+      key: "IMPORT_APP_FROM_FILE_MESSAGE",
+      value: "Drag and drop your file or upload from your computer",
+    },
+    {
+      key: "IMPORT_APP_FROM_GIT_MESSAGE",
+      value: "Import an application from its git repository using its SSH URL",
+    },
+    {
+      key: "IMPORT_FROM_GIT_REPOSITORY",
+      value: "Import from Git Repository",
```
Comment (IAmAnubhavSaini) [inline app/client/src/ce/constants/messages.test.ts:413]: Should be `Import from git repository`.

File: app/client/src/ce/constants/messages.test.ts [hunk 4]
```diff
@@ -386,3 +392,46 @@ describe("git-sync messages", () => {
...
+    },
+    {
+      key: "IMPORT_FROM_GIT_REPOSITORY_MESSAGE",
```
Comment (IAmAnubhavSaini) [inline app/client/src/ce/constants/messages.test.ts:416]: This key and value are not used anywhere. Remove it.

File: app/client/src/ce/constants/messages.test.ts [hunk 5]
```diff
@@ -386,3 +392,46 @@ describe("git-sync messages", () => {
...
+      value:
+        "While importing Appsmith will does not import the datasource credentials to prevent a breach. After a successfull import you can add the credentials manually so the application behaves normally!",
```
Comment (IAmAnubhavSaini) [inline app/client/src/ce/constants/messages.test.ts:418]: - Typo: use either will or does.
- Are credentials ever exported?

File: app/client/src/pages/Applications/ImportApplicationModal.tsx [hunk 6]
```diff
@@ -170,6 +176,29 @@ const CardWrapper = styled.div`
...
 
+const StatusbarWrapper = styled.div`
+  width: 252px;
+  height: 199px;
+  .cs-icon {
+    margin: auto;
+    border-radius: 50%;
+    width: 32px;
+    height: 32px;
+    background: ${Colors.GREY_4};
```
Comment (IAmAnubhavSaini) [inline app/client/src/pages/Applications/ImportApplicationModal.tsx:187]: Use CSS color variables.

File: app/client/src/pages/Applications/ImportApplicationModal.tsx [hunk 7]
```diff
@@ -242,7 +271,7 @@ function ImportApplicationModal(props: ImportApplicationModalProps) {
             applicationFile: file,
           }),
         );
-        onClose && onClose();
+        // onClose && onClose();
```
Comment (IAmAnubhavSaini) [inline app/client/src/pages/Applications/ImportApplicationModal.tsx:274]: This should be removed.

File: app/client/src/pages/Applications/ImportApplicationModal.tsx [hunk 8]
```diff
@@ -279,27 +309,48 @@ function ImportApplicationModal(props: ImportApplicationModalProps) {
...
+              uploadIcon="file-line"
+            />
+          </FileImportCard>
+          {isGitImportFeatureEnabled && <GitImportCard handler={onGitImport} />}
+        </Row>
+      )}
+      {importingApplication && (
+        <Row className="t-import-app-progress-wrapper">
+          <StatusbarWrapper className="t--importing-app-statusbar">
+            <Icon fillColor={Colors.GREY_800} name="file-line" />
```
Comment (IAmAnubhavSaini) [inline app/client/src/pages/Applications/ImportApplicationModal.tsx:342]: Use CSS color variables.

File: app/client/src/pages/Applications/ImportApplicationModal.tsx [hunk 9]
```diff
@@ -279,27 +309,48 @@ function ImportApplicationModal(props: ImportApplicationModalProps) {
...
+            <Text className="importing-app-name" type={TextType.P2}>
+              {appFileToBeUploaded?.file.name || "filename.JSON"}
```
Comment (IAmAnubhavSaini) [inline app/client/src/pages/Applications/ImportApplicationModal.tsx:344]: 1. `appFileToBeUploaded?.file?.name`
2. `"filename.json"`

**Review by IAmAnubhavSaini — CHANGES_REQUESTED** [review]

File: app/client/src/ce/constants/messages.test.ts (hunk 1 above)
Comment (haojin111) [reply app/client/src/ce/constants/messages.test.ts:398]: the text is fetched from figma design

File: app/client/src/ce/constants/messages.test.ts (hunk 2 above)
Comment (haojin111) [reply app/client/src/ce/constants/messages.test.ts:401]: the text is fetched from figma design

File: app/client/src/ce/constants/messages.test.ts (hunk 1 above)
Comment (IAmAnubhavSaini) [reply app/client/src/ce/constants/messages.test.ts:398]: @Debsourabh could you please update Figma design. I have left comments there too. https://www.figma.com/file/IOx0WPeZsChCaXwgVxU3LX/Import%2FExport-Design?node-id=1594%3A8197

**Review by IAmAnubhavSaini — APPROVED** [review]

**Pull request merged**